server/slack_notifier.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/applikatoni/applikatoni/deploy"
+	"github.com/applikatoni/applikatoni/models"
+
+	"log"
+	"net/http"
+	"text/template"
+)
+
+const slackSuccessSummaryTmplStr = `{{.GitHubRepo}} Successfully Deployed:
+{{.Username}} deployed {{.Branch}} on {{.Target}} :pizza:
+
+> {{.Comment}}
+<{{.GitHubUrl}}|View latest commit on GitHub>
+<{{.DeploymentURL}}|Open deployment in Applikatoni>`
+
+const slackFailSummaryTmplStr = `:rotating_light:  {{.GitHubRepo}} Deployed Failed :rotating_light: :
+{{.Username}} deployed {{.Branch}} on {{.Target}} :pizza:
+
+> {{.Comment}}
+<{{.GitHubUrl}}|View latest commit on GitHub>
+<{{.DeploymentURL}}|Open deployment in Applikatoni>`
+
+var slackSuccessSummaryTemplate = template.Must(template.New("slackSuccessSummary").Parse(slackSuccessSummaryTmplStr))
+var slackFailSummaryTemplate = template.Must(template.New("slackSuccessSummary").Parse(slackFailSummaryTmplStr))
+
+type slackMsg struct {
+	Text string `json:"text"`
+}
+
+func NotifySlack(db *sql.DB, deploymentId int, success bool) {
+	deployment, err := getDeployment(db, deploymentId)
+	if err != nil {
+		log.Printf("Could not find deployment with id %v, %s\n", deploymentId, err)
+		return
+	}
+
+	application, err := findApplication(deployment.ApplicationName)
+	if err != nil {
+		log.Printf("Could not find application with name %v, %s\n", deployment.ApplicationName, err)
+		return
+	}
+
+	target, err := findTarget(application, deployment.TargetName)
+	if err != nil {
+		log.Printf("Could not find target with name %v, %s\n", deployment.TargetName, err)
+		return
+	}
+
+	user, err := getUser(db, deployment.UserId)
+	if err != nil {
+		log.Printf("Could not find User with id %v, %s\n", deployment.UserId, err)
+		return
+	}
+
+	if target.SlackUrl == "" {
```

Comment:
You could move this up one paragraph so you don't need to load the user when you know that the target doesn't have Slack configured.